src/semaphore.rs

```diff
@@ -228,7 +228,7 @@ impl Future for AcquireArc {
                 }
                 None => {
                     // Wait on the listener.
-                    match &mut this.listener.take() {
+                    match &mut this.listener {
```

Comment:
This should probably just be `listener.take()`, without the `&mut`.

Reply:
We want to leave the listener in place if polling it returns `Pending` (in which case we return early at the `ready!`), no need to create a new one. (It was `take`n before #34 in order to `await` it, making the `async fn` lowering responsible for repeatedly polling it).

Reply:
Ah, good point.